### PR TITLE
os/bluestore: report error and quit correctly when disk error happens.

### PR DESCRIPTION
**Pull request by liupan1111**
Fixes: http://tracker.ceph.com/issues/21263
Signed-off-by: Pan Liu <wanjun.lp@alibaba-inc.com>

### CODE REVIEW

**Comment by liupan1111** [comment]
@liewegas, please help take a look, thanks.

**Comment by liewegas** [comment]
Also, do you mind making the same change to os/bluestore/KernelDevice.cc?  If error is != 0 and != EINTR, then assert.  Thanks!

File: src/os/bluestore/KernelDevice.cc [hunk 1]
```diff
@@ -342,6 +342,7 @@ void KernelDevice::_aio_thread()
 					 aio, max);
     if (r < 0) {
       derr << __func__ << " got " << cpp_strerror(r) << dendl;
+      assert(0 == "got unexpected error from io_getevents");
```
Comment (liupan1111) [inline src/os/bluestore/KernelDevice.cc]: @liewegas do you mean I should ignore EINTR here? If yes, I don't follow...

In bluestore/aio.cc, there is:

int aio_queue_t::get_next_completed(int timeout_ms, aio_t **paio, int max)
{
  io_event event[max];
  struct timespec t = {
    timeout_ms / 1000,
    (timeout_ms % 1000) * 1000 * 1000
  };

 ```
 int r = 0;
  do {
    r = io_getevents(ctx, 1, max, event, &t);
  } while (r == -EINTR);
```

So io_getevents will be called again if r == -EINTR, so I don't think the return value(r) could be -EINTR. Please correct me if I have anything wrong.

**Comment by liupan1111** [comment]
> Also, do you mind making the same change to os/bluestore/KernelDevice.cc? If error is != 0 and != EINTR, then assert. Thanks!

er... My change is now happened in os/bluestore/KernelDevice.cc ... do you mean FileJouranl.cc??

**Comment by liupan1111** [comment]
retest this please

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-09-07_15:42:04-rados-wip-sage-testing-2017-09-07-1015-distro-basic-smithi/

**Pull request merged**